src/collection.js

```diff
@@ -365,10 +377,32 @@ export default class Collection {
         "max_target.data.last_modified": String(at),
       },
     });
+    return changes;
+  }
 
+  /**
+   * @private
+   */
+  async getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    // If history doesn't have enough data, reject with an error
+    const oldestHistory = await this.findHistoryOldestTimestamp();
```

Comment:
When I look at the 2 requests in `listChangesBackTo` and `findHistoryOldestTimestamp()` I cannot see any problem about the requested range. In one we ask for `T[0]{1}` and the other one for `T[at..0]`. In both we have the oldest entry (`T[0]`).

So here we could use `changes[changes.length -1]` (not `[0]` as I said earlier, I had missed the sort).